src/renderer/angular/directives/commons/sk-tx-history.directive.js

```diff
@@ -0,0 +1,116 @@
+'use strict';
+
+function SKTxHistoryDirective($rootScope, $interval, RPCService, CommonService) {
+    'ngInject';
+
+    return {
+        restrict: 'E',
+        scope: {
+            tokenSymbol: '@'
+        },
+        link: (scope, element) => {
+            let publicKey = '0x' + $rootScope.wallet.getPublicKey();
+
+            let syncByWallet = () => {
+                RPCService.makeCall('syncTxHistoryByWallet', { walletId: $rootScope.wallet.id, publicKey });
+            };
+            syncByWallet();
+
+            scope.txList = [];
+            scope.tokenSymbol = scope.tokenSymbol ? scope.tokenSymbol.toUpperCase() : null;
+
+            let getTxStatusText = (tx) => {
+                let status = tx.txReceiptStatus;
+                let isSend = tx.from == publicKey;
+
+                if (status == 0) {
+                    return isSend ? 'Faild To Send' : 'Faild To Receive';
+                }
+                if (status == 1) {
+                    return isSend ? 'Sent' : 'Received';
+                }
+                return isSend ? 'Sending' : 'Sending';
+            };
+
+
+            let getTxStatusIcon = (tx) => {
+                let status = tx.txReceiptStatus;
+                let isSend = tx.from == publicKey;
+
+                if (status == 1) {
+                    return isSend ? 'sent' : 'receive';
+                }
+
+                if (status == 0) {
+                    return 'fail';
+                }
+
+                return 'clock';
+            };
+
+            let processTxHistoryList = (list) => {
+                return list.map((tx) => {
+                    let symbol = tx.tokenSymbol;
+                    tx.symbol = symbol ? symbol.toUpperCase() : 'ETH';
+                    tx.directionSign = publicKey == tx.from ? '- ' : '+ ';
+
+                    tx.externalLink = `https://etherscan.io/tx/${tx.hash}`;
+                    tx.statusText = getTxStatusText(tx);
+                    tx.statusIcon = getTxStatusIcon(tx);
```

Comment:
``getTxStatusText`` and ``getTxStatusIcon`` could be ``getTxStatus`` that returns object with ``iconName`` and ``text``.